compiler/modules/types/core.bal

```diff
@@ -815,9 +815,59 @@ function comparableNillableList(Context cx, SemType t1, SemType t2) returns bool
     }
     ComparableMemo memo = { semType1: t1, semType2: t2 };
     cx.comparableMemo.add(memo);
-    // SUBSET need to iterate members when tuples are supported
-    // following relies on the fact `listMemberType(cx, NIL, ()) = NEVER`
-    boolean result = comparable(cx, listMemberType(cx, t1, INT), listMemberType(cx, t2, INT));
+    boolean result;
+    var members1 = listAllMemberTypes(cx, t1);
+    var members2 = listAllMemberTypes(cx, t2);
+    // listAllMemberTypes return () if list type is not atomic
+    if members1 != () && members2 != () {
+        result = false;
+        int currentIndex = 0;
```

Comment:
Use shorter names for currentIndex/currentMemberRange/currentMemberTy.

Also be consistent about abbreviating. It doesn't make sense to have a name where `current` is unabbreviated more `Type` is abbreviated to `Ty`, since `cur` is the better abbreviation.